bin/npg_manifest4study

```diff
@@ -273,6 +291,21 @@ sub _uploaded_files {
     sprintf 'Failed to close the %s handle', $files_info_from_stdin ? 'STDIN' : 'pipe');
   ## use critic
 
+  if ($require_qc_complete and keys %{$files}) {
+    my @id_runs   = map { $files->{$_}->{id_run} } keys %{$files};
```

Comment:
```suggestion
    my @id_runs   = map { $_->{id_run} } values %{$files};
```

Reply:
Decided against because in a similar piece of code a bit lower had to explicitly select for keys that do not give a warning about undefined values when retrieving run id. Instead, produced a filtered list of keys, which is guaranteed to have file names and is then used twice.